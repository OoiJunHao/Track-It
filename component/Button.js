import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'

const Button = props => (
    <TouchableOpacity style={[styles.container, props.style]} onPress={props.onPress}>
        <Text style={styles.buttonText}>{props.children}</Text>
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#2980b9',
        paddingVertical: 15,
        marginBottom: 10,
        borderRadius: 15
    },
    buttonText: {
        textAlign: 'center',
        color: 'rgb(108,108,108)',
        fontWeight: '700'
    }
});

export default Button