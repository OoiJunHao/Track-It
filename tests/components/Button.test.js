import React from 'react';
import { render, cleanup, fireEvent } from 'react-native-testing-library';
import Button from '../../component/Button';

afterEach(cleanup);

describe('button', () => {
    it('should match the snapshot', () => {
        const rendered = render(<Button children={'abacaba'} />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

    it('should properly render the text', () => {
        const rendered = render(<Button children={'abacaba'} />);
        const textComponent = rendered.getByTestId('text');
        expect(textComponent.props.children).toEqual('abacaba');
    });

    it('should fire onSubmit events', () => {
        const onSubmit = jest.fn();
        const rendered = render(<Button onPress={onSubmit} />);
        const buttonComponent = rendered.getByTestId('submit');

        fireEvent(buttonComponent, 'press');

        expect(onSubmit).toHaveBeenCalled();
    });
});
