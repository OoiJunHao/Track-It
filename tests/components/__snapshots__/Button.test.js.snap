// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match the snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2980b9",
      "borderRadius": 15,
      "marginBottom": 10,
      "opacity": 1,
      "paddingVertical": 15,
    }
  }
  testID="submit"
>
  <Text
    style={
      Object {
        "color": "rgb(108,108,108)",
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
    testID="text"
  >
    abacaba
  </Text>
</View>
`;
